internal/pkg/danger/buf.go

```diff
@@ -33,7 +33,11 @@ func (b *Buf) Reset() {
 }
 
 func (b *Buf) grow(n int) {
-	buf := make([]byte, len(b.buf), 2*cap(b.buf)+n)
+	newCap := 2*cap(b.buf) + n
+	if newCap < 0 {
+		panic("danger.Buf.grow: size computation overflow")
```

Comment:
This is an internal method that only has 2 calls in this file, one that already explicitly handles a negative count the same way bytes.Buffer does in Go itself https://cs.opensource.google/go/go/+/refs/tags/go1.24.0:src/bytes/buffer.go;l=164-170

https://github.com/elastic/fleet-server/blob/ba68a24383abfa328cce0cc3eddce1b73b497d8a/internal/pkg/danger/buf.go#L41-L48

And another that uses an explicitly non-negative constant: https://github.com/elastic/fleet-server/blob/ba68a24383abfa328cce0cc3eddce1b73b497d8a/internal/pkg/danger/buf.go#L66-L74

This change seems unnecessary.